docs/_docs/sql-reference/explain-help-with-performance-problems.adoc

```diff
@@ -0,0 +1,495 @@
+But plan can shows:
+[source,sql]
+----
+Project
+  Exchange
+    Sort
+        collation: [NAME ASC]
+      IndexScan
+          table: PUBLIC.T
+          index: EMP_SALARY_NAME_IDX
+          filters: =(SALARY, 1)
+----
+We can see - additional **Sort** operator which is redundant here.
+A bit query refactoring can help to avoid excessive sort:
+[source,sql]
+----
+SELECT dept_id FROM emp WHERE salary = 1 ORDER BY salary, name;
+----
+And the plan becomes as follows:
+[source,sql]
+----
+Project
+  Exchange
+    IndexScan
+        table: PUBLIC.T
+        index: EMP_SALARY_NAME_IDX
+        filters: =(SALARY, 1)
+----
+== Colocated data miss
+In many cases it is beneficial to colocate different entries if they are often accessed together.
+In this way, multi-entry queries are executed on the same node (where the columns are stored). This concept is known as affinity colocation.
+Let`s consider schema and follow query:
+[source,sql]
+----
+-- explicitly colocated by (dept_id)
+CREATE TABLE emp(dept_id INTEGER, name VARCHAR, salary INTEGER, PRIMARY KEY(dept_id, name)) COLOCATE BY(dept_id);
+
+-- miss COLOCATE BY and implicitly colocated by (id, name)
+CREATE TABLE dept(id INTEGER, name VARCHAR, PRIMARY KEY(id, name));
+
+SELECT emp.name, dept.name FROM emp JOIN dept ON emp.dept_id = dept.id;
+----
+Expect colocated join here, but plan shows that it`s not true:
+[source,sql]
+----
+Exchange
+  Project
+    HashJoin
+      TableScan
+          table: PUBLIC.EMP
+      Exchange
+          distribution: affinity
+        TableScan
+            table: PUBLIC.DEPT
+----
+We can see additional **Exchange** operator, which indicates that rows from one node are transferred into another.
+After changing 'lost' colocation:
+[source,sql]
+----
+-- explicitly colocated by (dept_id)
+CREATE TABLE emp(dept_id INTEGER, name VARCHAR, salary INTEGER, PRIMARY KEY(dept_id, name)) COLOCATE BY(dept_id);
+
+-- explicitly colocated by (id)
+CREATE TABLE dept(id INTEGER, name VARCHAR, PRIMARY KEY(id, name)) COLOCATE BY(id);
+
+SELECT emp.name, dept.name FROM emp JOIN dept ON emp.dept_id = dept.id;
+----
+Possible plan will show correct colocated join:
+[source,sql]
+----
+Exchange
+  Project
+    HashJoin
+      TableScan
+          table: PUBLIC.EMP
+      TableScan
+          table: PUBLIC.DEPT
+----
+== Optimized plans
+Some queries can be optimized to use more optimal plans which brings performance speed up. For example, plan for:
+[source,sql]
+----
+SELECT COUNT(*) FROM emp;
+----
+Can look like:
+[source,sql]
+----
+SelectCount
+    table: [PUBLIC, EMP]
+    est: (rows=43)
+----
+But there are numerous cases where such optimization is not applicable (transactional guarantees or some other restrictions), in such a case plan can be a bit different and more time consumable for execution.
+The same query as above, but with explicit transaction may produce a different plan like:
+[source,sql]
+----
+Project
+  ReduceSortAggregate
+    Exchange
+      MapSortAggregate
```

Comment:
It is not clear why SortAggregate is used? 
Because TableScan has no collation and no Sort node.

Reply:
The goal of this part is to show that optimized plans can evaluate into less productive one. In your case i will need to describe how MapSortAggregate work, but it is not a case here.